testing/py-ycmd/APKBUILD

```diff
@@ -0,0 +1,390 @@
+	$pkgname-$pkgver-bottle.zip::https://github.com/bottlepy/$_bottlepkgname/archive/$_bottle_gitrev.zip
+	$pkgname-$pkgver-python-frozendict.zip::https://github.com/slezica/$_frozendictpkgname/archive/$_frozendict_gitrev.zip
+	$pkgname-$pkgver-gocode.zip::https://github.com/nsf/$_gocodepkgname/archive/$_gocode_gitrev.zip
+	$pkgname-$pkgver-godef.zip::https://github.com/Manishearth/$_godefpkgname/archive/$_godef_gitrev.zip
+	$pkgname-$pkgver-python-future.zip::https://github.com/PythonCharmers/$_futurepkgname/archive/$_future_gitrev.zip
+	$pkgname-$pkgver-racerd.zip::https://github.com/jwilm/$_racerdpkgname/archive/$_racerd_gitrev.zip
+	$pkgname-$pkgver-requests.zip::https://github.com/requests/$_requestspkgname/archive/$_requests_gitrev.zip
+	$pkgname-$pkgver-waitress.zip::https://github.com/Pylons/$_waitresspkgname/archive/$_waitress_gitrev.zip
+	$pkgname-$pkgver-nrefactory.zip::https://github.com/icsharpcode/$_nrefactorypkgname/archive/$_nrefactory_gitrev.zip
+	$pkgname-$pkgver-cecil.zip::https://github.com/jbevain/$_cecilpkgname/archive/$_cecil_gitrev.zip
+	fix-max-line-length.patch
+	prepare-shutdown_test.patch"
+builddir="$srcdir/"$_pkgname-$_gitrev
+subpackages="py2-$_pkgname:_py2 py3-$_pkgname:_py3 $pkgname-doc"
+
+unpack() {
+	default_unpack
+	mv $_jedipkgname-$_jedi_gitrev/* $builddir/third_party/$_jedipkgname/
+	mv $_omnisharppkgname-$_omnisharp_gitrev/* $builddir/third_party/OmniSharpServer/
+	mv $_bottlepkgname-$_bottle_gitrev/* $builddir/third_party/$_bottlepkgname/
+	mv $_frozendictpkgname-$_frozendict_gitrev/* $builddir/third_party/frozendict/
+	mv $_gocodepkgname-$_gocode_gitrev/* $builddir/third_party/$_gocodepkgname/
+	mv $_godefpkgname-$_godef_gitrev/* $builddir/third_party/$_godefpkgname/
+	mv $_futurepkgname-$_future_gitrev/* $builddir/third_party/$_futurepkgname/
+	mv $_racerdpkgname-$_racerd_gitrev/* $builddir/third_party/$_racerdpkgname/
+	mv $_requestspkgname-$_requests_gitrev/* $builddir/third_party/$_requestspkgname/
+	mv $_waitresspkgname-$_waitress_gitrev/* $builddir/third_party/$_waitresspkgname/
+	mv $_nrefactorypkgname-$_nrefactory_gitrev/* $builddir/third_party/OmniSharpServer/$_nrefactorypkgname/
+	mv $_cecilpkgname-$_cecil_gitrev/* $builddir/third_party/OmniSharpServer/$_cecilpkgname/
+}
+
+_getpythonver(){
+	echo $($1 -c 'import sys; print("%i.%i" % (sys.version_info.major, sys.version_info.minor))')
+}
+
+_pybuild() {
+	local python="$1"
+	cp -a "$builddir" "${builddir}_$python"
+	cd "${builddir}_$python"
+	#[[ "$python" == "python3" ]] && sed -i -e "s|python3|python|g" cpp/ycm/CMakeLists.txt
+	sed -i -e "s|EXTERNAL_LIBCLANG_PATH \${TEMP}|EXTERNAL_LIBCLANG_PATH $(ls /usr/lib/libclang.so.*)|g" cpp/ycm/CMakeLists.txt
+	find "$srcdir" -type d -print0 | xargs -0 chmod 755
+	local arch_configure=""
+	[[ -n "$_e_c"    && "$_e_c" == "1" ]]    && arch_configure="$arch_configure --system-libclang --clang-completer"
```

Comment:
Instead of all of this, why not just build `$_arch_configure` as you enable the options outside `_pybuild()`?

Reply:
That is a valid suggestion.  I did it this way because old habit or because other package systems set up configure options in a configure() phase but APKBUILD does it in the build phase.  It would put the configuration in global scope so before all phases at the very top near lines 12-20.